src/hooks/transaction/extrinsics/lib.ts

```diff
@@ -67,13 +72,102 @@ const getLibExtrinsic = async (params: LibActions): Promise<ExtrinsicData> => {
+      const depositAmount = args[1];
+
+      const transferExtrinsic = window.bridge.tokens.transfer(proxyAccount.toString(), depositAmount);
+
+      const strategyDepositExtrinsic = (await window.bridge.loans.lend(...args)).extrinsic;
+      const proxiedStrategyDepositExtrinsic = proxifyExtrinsic(proxyAccount, strategyDepositExtrinsic);
+
+      if (isIdentitySet) {
+        const batchedExtrinsics = window.bridge.transaction.buildBatchExtrinsic([
+          transferExtrinsic.extrinsic,
+          proxiedStrategyDepositExtrinsic
+        ]);
+
+        return { extrinsic: batchedExtrinsics };
+      } else {
+        const identityLockAmount = new MonetaryAmount(GOVERNANCE_TOKEN, 26);
+        const identityLockAmountTransferExtrinsic = window.bridge.tokens.transfer(
+          proxyAccount.toString(),
+          identityLockAmount
+        );
+        const strategyAccountIdentity = {
+          additional: [[{ Raw: 'strategyType' }, { Raw: strategyType }]]
+        };
+        const setIdentityExtrinsic = window.bridge.api.tx.identity.setIdentity(strategyAccountIdentity);
+        const proxiedSetIdentityExtrinsic = proxifyExtrinsic(proxyAccount, setIdentityExtrinsic);
+
+        const batchedExtrinsicsWithIdentity = window.bridge.transaction.buildBatchExtrinsic([
+          identityLockAmountTransferExtrinsic.extrinsic,
+          proxiedSetIdentityExtrinsic,
+          transferExtrinsic.extrinsic,
+          proxiedStrategyDepositExtrinsic
+        ]);
+
+        return { extrinsic: batchedExtrinsicsWithIdentity };
+      }
+    }
+
+    case Transaction.STRATEGIES_WITHDRAW: {
+      const primaryAccount = window.bridge.account;
+      if (!primaryAccount) {
+        throw new Error('Strategy primary account not found.');
+      }
+
+      const [, proxyAccount, ...args] = params.args;
+      const withdrawalAmount = args[1];
+
+      const strategyWithdrawalExtrinsic = (await window.bridge.loans.withdraw(...args)).extrinsic;
+      const proxiedStrategyWithdrawExtrinsic = proxifyExtrinsic(proxyAccount, strategyWithdrawalExtrinsic);
+
+      const transferExtrinsic = window.bridge.tokens.transfer(primaryAccount.toString(), withdrawalAmount).extrinsic;
+      const proxiedTransferExtrinsic = proxifyExtrinsic(proxyAccount, transferExtrinsic);
+
+      const batchExtrinsic = window.bridge.transaction.buildBatchExtrinsic([
+        proxiedStrategyWithdrawExtrinsic,
+        proxiedTransferExtrinsic
+      ]);
+
+      return { extrinsic: batchExtrinsic };
+    }
+
     case Transaction.STRATEGIES_ALL_WITHDRAW: {
-      const [underlyingCurrency] = params.args;
-      return window.bridge.loans.withdrawAll(underlyingCurrency);
+      const primaryAccount = window.bridge.account;
+      if (!primaryAccount) {
+        throw new Error('Primary account not found.');
+      }
+
+      const [, proxyAccount, ...args] = params.args;
+      const withdrawalAmount = args[1];
```

Comment:
cant you just destruct this from the array in the live above as  you did with the `proxyAccount`?

Reply:
This applies for the other occurrences